plugins/scanners/dos/src/main/kotlin/DosScanner.kt

```diff
@@ -233,15 +234,25 @@ class DosScanner internal constructor(
     }
 }
 
-private fun Collection<Package>.getDosPurls(provenance: Provenance = UnknownProvenance): List<String> {
+private fun Collection<Package>.getDosPackages(provenance: Provenance = UnknownProvenance): List<PackageInfo> {
     val extras = provenance.toPurlExtras()
 
     return when (provenance) {
         is RepositoryProvenance -> {
             // Maintain the VCS path to get the "bookmarking" right for the file tree in the package configuration UI.
-            map { it.id.toPurl(extras.qualifiers, it.vcsProcessed.path) }
+            map {
+                PackageInfo(
+                    purl = it.id.toPurl(extras.qualifiers, it.vcsProcessed.path),
```

Comment:
Out of scope: Is this a bug taking the purl in this way? (I wonder if `it.purl` should be used instead)

Reply:
See the comment in line 242.

Reply:
Indeed purls from curations are currently discarded. I guess we could implement something here to not discard them, but at the same time ensure that they contain the required extra data about the provenance.